apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts

```diff
@@ -126,16 +111,19 @@ onDocumentReady(() => {
     }
   });
 
-  if (!questionSettingsForm || !saveButton) return;
-  saveButtonEnabling(questionSettingsForm, saveButton);
+  if (questionSettingsForm && saveButton) {
+    saveButtonEnabling(questionSettingsForm, saveButton);
+  }
 
+  updateWorkspaceOptionsValidation();
   showWorkspaceOptionsButton?.addEventListener('click', () => {
     workspaceOptions?.removeAttribute('hidden');
     showWorkspaceOptionsButton.setAttribute('hidden', 'true');
+    workspaceOptionsShown = true;
+    updateWorkspaceOptionsValidation();
   });
```

Comment:
I think this default behavior is a little bit strange that the checkbox is not checked even though the default for `rewrite_url` is `true`.  Can we add a line to the event listener here to check the `rewrite_url` checkbox when we click the button to add workspace settings?

```suggestion
  showWorkspaceOptionsButton?.addEventListener('click', () => {
    workspaceOptions?.removeAttribute('hidden');
    showWorkspaceOptionsButton.setAttribute('hidden', 'true');
    workspaceOptionsShown = true;
    document.querySelector<HTMLInputElement>('#workspace_rewrite_url')?.setAttribute('checked', 'true');
    updateWorkspaceOptionsValidation();
  });
```

Reply:
You are correct, this is working as it should. Thanks!